docs/notebooks/multi_objective_ehvi.pct.py

```diff
@@ -0,0 +1,201 @@
+# -*- coding: utf-8 -*-
+# %% [markdown]
+# # Multi-objective optimization with Expected HyperVolume Improvement
+
+# %%
+import math
+import gpflow
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+from util.plotting import plot_bo_points, plot_function_2d, plot_mobo_history, plot_mobo_points_in_obj_space
+
+# %%
+import trieste
+from trieste.acquisition.function import ExpectedHypervolumeImprovement
+from trieste.acquisition.rule import OBJECTIVE
+from trieste.data import Dataset
+from trieste.models import create_model
+from trieste.models.model_interfaces import ModelStack
+from trieste.space import Box
+from trieste.utils.multi_objectives import VLMOP2
+from trieste.utils.pareto import Pareto, get_reference_point
+from trieste.acquisition.rule import EfficientGlobalOptimization
+
+np.random.seed(1793)
+tf.random.set_seed(1793)
+
+# %% [markdown]
+# ## Describe the problem
+#
+# In this tutorial, we provide multi-objective optimization example using the expected hypervolume improvement acquisition function.
+# We consider the VLMOP2 function --- a synthetic benchmark problem with two objectives. We start by defining the problem parameters.
+
+
+# %%
+vlmop2 = VLMOP2().objective()
+observer = trieste.utils.objectives.mk_observer(vlmop2, OBJECTIVE)
+
+# %%
+mins = [-2, -2]
+maxs = [2, 2]
+search_space = Box(mins, maxs)
+num_objective = 2
+
+# %% [markdown]
+# Let's randomly sample some initial data from the observer ...
+
+# %%
+num_initial_points = 20
+initial_query_points = search_space.sample(num_initial_points)
+initial_data = observer(initial_query_points)
+
+# %% [markdown]
+# ... and visualise the data across the design space.
+
+# %%
+_, ax = plot_function_2d(
+    vlmop2,
+    mins,
+    maxs,
+    grid_density=100,
+    contour=True,
+    title=["Obj 1", "Obj 2"],
+    figsize=(12, 6),
+    colorbar=True,
+    xlabel="$X_1$",
+    ylabel="$X_2$",
+)
+plot_bo_points(initial_query_points, ax=ax[0, 0], num_init=num_initial_points)
+plot_bo_points(initial_query_points, ax=ax[0, 1], num_init=num_initial_points)
+plt.show()
+
+# %% [markdown]
+# ... and in the objective space
+
+# %%
+plot_mobo_points_in_obj_space(initial_data[OBJECTIVE].observations)
+plt.show()
+
+
+# %% [markdown]
+# ## Modelling the two functions
+#
+# In this example we model the two objective functions individually with their own Gaussian process models. For problems where the objective functions are similar it may make sense to build a joint model rather than a model stack.
```

Comment:
This is the only place in the docs where we use `ModelStack`. I think that it's quite important to describe it fully. I would even explain (in one our 2 sentences) why we don't use a dictionary as in`/docs/notebooks/inequality_constraints.pct.py`

Reply:
probably worth adding an entry to the FAQ on how to handle multiple models - but probably not part of this PR